lib/hawkular/inventory/entities.rb

```diff
@@ -0,0 +1,217 @@
+# It contains class definitions that are used by the inventory REST client
+module Hawkular::Inventory
+  # A Basic inventory entity with id, name, path and optional properties
+  class BaseEntity
+    # @return [String] Full path of the entity
+    attr_reader :path
+    # @return [String] Name of the entity
+    attr_reader :name
+    # @return [String] Name of the entity
+    attr_reader :id
+    # @return [String] Feed this entity belongs to (or nil in case of a feedless entity)
+    attr_reader :feed
+    # @return [String] Name of the environment for this entity
+    attr_reader :env
+    # @return [String] Properties of this entity
+    attr_reader :properties
+
+    def initialize(hash)
+      @id = hash['id']
+      @path = hash['path']
+      @name = hash['name'] || @id
+      @properties = hash['properties'] || {}
+      @_hash = hash.dup
```

Comment:
Right, it's there from the Libor's times. To be honest, I don't know. What it does is a deep copy of the hash that is passed to the constructor. This looks like a good idea, because if you do this

```ruby
def foo(hash)
  hash['foo'] = 0
  hash
end
my_hash = { 'foo' => 1 }
foo(my_hash)
puts my_hash .. will print.. { 'foo' => 0 }
```

I guess, the `@_hash` is there to make the `to_h` method faster (+less code to write)